test/test_images/latencymako/aggregator.go

```diff
@@ -30,49 +27,59 @@ import (
 	"github.com/golang/protobuf/ptypes"
 	"github.com/golang/protobuf/ptypes/timestamp"
 
-	pb "knative.dev/eventing/test/test_images/latencymako_aggregator/event_state"
-	"knative.dev/pkg/signals"
+	pb "knative.dev/eventing/test/test_images/latencymako/event_state"
 	"knative.dev/pkg/test/mako"
 )
 
 const listenAddr = ":10000"
 const maxRcvMsgSize = 1024 * 1024 * 10
 
-// flags for the image
-var (
-	verbose       bool
-	expectRecords uint
-)
-
-var fatalf = log.Fatalf
-
-func init() {
-	flag.BoolVar(&verbose, "verbose", false, "Enable verbose logging")
-	flag.UintVar(&expectRecords, "expect-records", 1, "Number of expected events records before aggregating data.")
-}
-
-// aggregation of received events records
+// thread-safe events recording map
 type eventsRecord struct {
 	sync.RWMutex
-	pb.EventsRecord
+	*pb.EventsRecord
 }
 
-var (
-	sentEvents     eventsRecord
-	acceptedEvents eventsRecord
-	failedEvents   eventsRecord
-	receivedEvents eventsRecord
-)
+type aggregatorExecutor struct {
```

Comment:
The context of the test is now wrapped inside executor structs, so that the dependencies of each component are clearly defined. There is 1 `aggregatorExecutor` and 1 `senderReceiverExecutor`. This also avoids relying on package-scoped variables.